devbox/src/syncer/AgentReadWriteActor.scala

```diff
@@ -183,7 +183,14 @@ class AgentReadWriteActor(agent: AgentApi,
   def getRpcFor(msg: SyncFiles.Msg, buf: ByteBuffer): Option[Rpc] = {
     msg match{
       case SyncFiles.Complete() => Some(Rpc.Complete())
-      case SyncFiles.RemoteScan(paths) => Some(Rpc.FullScan(paths))
+      case SyncFiles.RemoteScan(paths, forceIncludes) =>
+        // convert each MutablePathSet to a Seq, which is what the RPC message expects
+        val seqs = for (pathset <- forceIncludes) yield {
+          pathset.walk(Seq.empty).foldLeft(Seq.empty[os.SubPath]) { (ps: Seq[os.SubPath], segs: IndexedSeq[String]) =>
+            ps :+ os.SubPath.apply(segs: IndexedSeq[String])
```

Comment:
I think this `foldLeft` is accidentally quadratic, since `Seq.empty` is a `List` with an O(n) `:+` operation

Reply:
This should be equivalent to the following `.map` call I think

```scala
pathset.walk(Seq.empty).map(segs => os.SubPath.apply(segs: IndexedSeq[String])).to(Vector)
```